examples/profiling/primitive_instrumentation.cpp

```diff
@@ -0,0 +1,205 @@
+        if (counter_values.empty())
+        {
+            // Iterate through the last parts of performance counter names
+            for (auto const& counter_name_last_part : counter_name_last_parts)
+            {
+                // Construct the name of the counter
+                std::string counter_name("/phylanx/primitives/" +
+                    tags.primitive + "/" + counter_name_last_part);
+                // The actual performance counter
+                hpx::performance_counters::performance_counter counter(
+                    counter_name, locality_id);
+                counter_values.push_back(
+                    counter.get_counter_values_array(hpx::launch::sync, false)
+                        .values_);
+            }
+        }
+
+        std::vector<std::int64_t> data(counter_name_last_parts.size());
+        for (int i = 0; i < counter_values.size(); ++i)
+        {
+            data[i] = counter_values[i][tags.sequence_number];
+        }
+
+        result[name] = data;
+    }
+
+    return result;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int hpx_main()
+{
+    // compile the given code
+    phylanx::execution_tree::compiler::function_list snippets;
+
+    auto lra = phylanx::execution_tree::compile(
+        phylanx::ast::generate_ast(lra_code), snippets);
+
+    // print instrumentation information, if enabled
+    auto entries =
+        hpx::agas::find_symbols(hpx::launch::sync, "/phylanx/*");
+
+    // LRA arguments
+
+    blaze::DynamicMatrix<double> v1{ { 15.04, 16.74 },{ 13.82, 24.49 },
+    { 12.54, 16.32 },{ 23.09, 19.83 },{ 9.268, 12.87 },{ 9.676, 13.14 },
+    { 12.22, 20.04 },{ 11.06, 17.12 },{ 16.3 , 15.7 },{ 15.46, 23.95 },
+    { 11.74, 14.69 },{ 14.81, 14.7 },{ 13.4 , 20.52 },{ 14.58, 13.66 },
+    { 15.05, 19.07 },{ 11.34, 18.61 },{ 18.31, 20.58 },{ 19.89, 20.26 },
+    { 12.88, 18.22 },{ 12.75, 16.7 },{ 9.295, 13.9 },{ 24.63, 21.6 },
+    { 11.26, 19.83 },{ 13.71, 18.68 },{ 9.847, 15.68 },{ 8.571, 13.1 },
+    { 13.46, 18.75 },{ 12.34, 12.27 },{ 13.94, 13.17 },{ 12.07, 13.44 } };
+
+    blaze::DynamicVector<double> v2{ 1, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0,
+        1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1 };
+    auto x = phylanx::ir::node_data<double>{ v1 };
+    auto y = phylanx::ir::node_data<double>{ v2 };
+    auto alpha = phylanx::ir::node_data<double>{ 1e-5 };
+    bool enable_output = true;
+
+    // Time execution
+    hpx::util::high_resolution_timer t;
+
+    // Evaluate LRA using the read data
+    auto result = lra(x, y, alpha);
+
+    // CSV Header
+    std::cout << "primitive_instance" << ","
+        << "count" << ","
+        << "time" << ","
+        << "direct_count" << ","
+        << "direct_time"
+        << std::endl;
+
+    // List of existing primitive instances
+    std::vector<std::string> existing_primitive_instances;
+    // Iterate over all primitive types
+    for (auto const& pattern :
```

Comment:
Why do you iterate over all known patterns? Wouldn't it be sufficient to call `hpx::agas::find_symbols()` once? This should return all known primitive names that are registered at this point.

Reply:
Fixed. Made other changes that need review. Also, I also added `/wd4552` to `Phylanx_SetupCompilerFlags.cmake` so that MSVC stops giving us this warning:
```
warning C4552: '*': operator has no effect; expected operator with side-effect
```

Reply:
At some point a discrepancy between `function` and `call-function` was introduced and somehow all tests have been passing since but it broke my code. We need to somehow test for this.